packages/suite/src/hooks/wallet/useStakeEthForm.ts

```diff
@@ -0,0 +1,332 @@
+
+    const onCryptoAmountChange = useCallback(
+        async (amount: string) => {
+            setIsAmountForWithdrawalWarningShown(false);
+            if (!fiatRates || !fiatRates.current) return;
+
+            const fiatValue = toFiatCurrency(amount, localCurrency, fiatRates.current.rates);
+            setValue('setMaxOutputId', undefined, { shouldDirty: true });
+            setValue(FIAT_INPUT, fiatValue || '', { shouldValidate: true });
+            setValue(OUTPUT_AMOUNT, amount || '', { shouldDirty: true });
+            await composeRequest(CRYPTO_INPUT);
+        },
+        [composeRequest, fiatRates, localCurrency, setValue],
+    );
+
+    const onFiatAmountChange = useCallback(
+        (amount: string) => {
+            setValue('setMaxOutputId', undefined, { shouldDirty: true });
+            setIsAmountForWithdrawalWarningShown(false);
+            if (!fiatRates || !fiatRates.current) return;
+
+            const cryptoValue = fromFiatCurrency(
+                amount,
+                localCurrency,
+                fiatRates.current.rates,
+                network.decimals,
+            );
+            setValue(CRYPTO_INPUT, cryptoValue || '', { shouldDirty: true, shouldValidate: true });
+            setValue(OUTPUT_AMOUNT, cryptoValue || '', {
+                shouldDirty: true,
+            });
+            composeRequest(FIAT_INPUT);
+        },
+        [composeRequest, fiatRates, localCurrency, network.decimals, setValue],
+    );
+
+    const setRatioAmount = useCallback(
+        async (divisor: number) => {
+            setValue('setMaxOutputId', undefined, { shouldDirty: true });
+            clearErrors([FIAT_INPUT, CRYPTO_INPUT]);
+            setIsAmountForWithdrawalWarningShown(false);
+
+            const amount = new BigNumber(account.formattedBalance)
+                .dividedBy(divisor)
+                .decimalPlaces(network.decimals)
+                .toString();
+
+            await onCryptoAmountChange(amount);
+            setValue(CRYPTO_INPUT, amount, { shouldDirty: true, shouldValidate: true });
+        },
+        [account.formattedBalance, clearErrors, network.decimals, onCryptoAmountChange, setValue],
+    );
+
+    const setMax = useCallback(async () => {
+        setValue('setMaxOutputId', 0, { shouldDirty: true });
+        clearErrors([FIAT_INPUT, CRYPTO_INPUT]);
+        await composeRequest(CRYPTO_INPUT);
+        setIsAmountForWithdrawalWarningShown(true);
+    }, [clearErrors, composeRequest, setValue]);
+
+    // TODO: Add more extra fee to ensure tx success when staking logic is implemented
+    const composedFee = useMemo(() => {
+        const transactionInfo = composedLevels?.[selectedFee];
+        return transactionInfo !== undefined && transactionInfo.type !== 'error'
+            ? new BigNumber(fromWei(transactionInfo.fee))
+            : new BigNumber('0');
+    }, [composedLevels]);
+
+    const isComposedFeeGreaterThanMin = composedFee.gt(MIN_ETH_FOR_FEES);
```

Comment:
We modified original issue. 0.03 ETH reserve should be for claiming and unstaking.

So the Max should do `balance - 0.03 - current fee`

Reply:
https://github.com/everstake/trezor-suite/pull/3/commits/1d1721ce124e600239df1fd8da961e4950bf2d3b